cni/pkg/repair/netns.go

```diff
@@ -104,27 +99,22 @@ func getPodNetNs(pod *corev1.Pod) (string, error) {
 	}
 	oldest := uint64(math.MaxUint64)
 	best := ""
-	// We will iterate over all processes. Our goal is to find a process with the same network ID as we found above.
+
+	// We will iterate over all processes. Our goal is to find a process whose namespace has a veth with an IP matching the pod.
 	// There should be 1 or 2 processes that match: the pause container should always be there, and the istio-validation *might*.
 	// We want the pause container, as the istio-validation one may exit before we are done.
 	// We do this by detecting the longest running process. We could look at `cmdline`, but is likely more reliable to weird platforms.
 	for _, p := range procs {
 		ns := getPidNamespace(p.PID)
-		fd, err := unix.Open(ns, unix.O_RDONLY, 0)
-		if err != nil {
-			// Not uncommon, many processes are transient and we have a TOCTOU here.
-			// No problem, must not be the one we are after.
-			log.Debugf("failed to open pid %v: %v", p.PID, err)
-			continue
-		}
-		id, err := netlink.GetNetNsIdByFd(fd)
-		_ = unix.Close(fd)
+		gip, err := runInNS(ns, func() (string, error) {
+			return getInterfaceAddr("eth0")
```

Comment:
eth0 seems equally implementation specific as /32 route?

Reply:
Yeah, this is what we found was the case in the platforms we tried, but we could just check each interface to be safe.

Reply:
Would prefer to check each, agree interface names aren't reliable.